datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -95,7 +110,50 @@ fn get_valid_types(
         TypeSignature::VariadicAny => {
             vec![current_types.to_vec()]
         }
+
         TypeSignature::Exact(valid_types) => vec![valid_types.clone()],
+        TypeSignature::ArrayAppendLikeSignature => {
```

Comment:
Have a specialized checking for array append at the end. And I think we will need specialized check for other pattern too